### DISCUSSION TITLE
Checkbox off value

### DISCUSSION THREAD

**Discussion by kirkegaard** [react-hook-form/react-hook-form#1832] [Q&A] [answered]
Is it possible to set an off value for a checkbox? 
I tried the trick with having a hidden input above with the same name but it seems to return false when unchecked. I could do a check in my submit function but it seems wrong.
I might have missed this in the docs but could anyone point me to where i find it?

**Comment by bluebill1049** [comment, marked as answer]
interesting one... maybe you will need to use `Controller` to parse the value.  take a look at this example: https://codesandbox.io/s/react-hook-form-parse-and-format-textarea-furtc maybe it's helpful.

**Comment by kirkegaard** [comment]
Aaah `Controller` works. Its a bit wonky but this is what i ended up doing. Wonder if theres a better way?

```javascript
<Controller
  name="mycheckbox"
  control={control}
  defaultChecked={true}
  defaultValue="on"
  className="form-check-input"
  as={<input type="checkbox" />}
  onChange={([event]) =>
    (event.target.value = event.target.checked
      ? "on"
      : "off")
  }
/>
```

**Reply by bluebill1049** [reply]
looks good to me.

**Reply by kirkegaard** [reply]
Thank you for your help! :) Good call on using `Controller`!

**Reply by bluebill1049** [reply]
In v6, controller usage will be better with render props.

**Comment by kirkegaard** [comment]
I just updated to `6.11.0` and this stopped working. Seems like i cant update value on the change event.
I tried moving the event into the render prop, but without any luck. Any idea on how to make this work again? :)

```js
<Controller
  name="mycheckbox"
  control={control}
  defaultChecked={true}
  defaultValue="on"
  render={(props) => (
    <input
      type="checkbox"
      {...props}
      onChange={(event) =>
        (event.target.value = event.target.checked
          ? "on"
          : "off")
      }
    />
  )}
/>
```

**Reply by bluebill1049** [reply]
can you share a codesandbox?

**Reply by kirkegaard** [reply]
Sure thing https://codesandbox.io/s/mutable-night-finkn

I might have misunderstod how controlled and rendered works so please bare with me :)

**Reply by bluebill1049** [reply]
<img width="445" alt="Screen Shot 2020-11-26 at 6 51 39 pm" src="https://user-images.githubusercontent.com/10513364/100322463-6d896000-3018-11eb-8037-aa075d39ce7b.png">

your controller example never sends any value to hook form tho. Make sure you are using the onChange prop form render.
